src/common/models/ICUHeadroom.ts

```diff
@@ -0,0 +1,230 @@
+import { nonNull } from 'common/utils';
```

Comment:
I refactored all of the ICU Headroom code into a new file since it's pretty sizable at this point.

I also basically rewrote it from scratch. Sorry.